common/zoom/user.ts

```diff
@@ -100,37 +106,55 @@ const createZoomUser = async (student: Pick<student, 'id' | 'firstname' | 'lastn
     return newUser;
 };
 
-async function getZoomUser(email: string): Promise<ZoomUser | null> {
+async function getZoomUser({ id, email, zoomUserId }: GetZoomUserArgs): Promise<ZoomUser | null> {
     assureZoomFeatureActive();
-
     const { access_token } = await getAccessToken();
-    const response = await zoomRetry(
-        () =>
-            fetch(`${zoomUserApiUrl}/${email}`, {
-                method: 'GET',
-                headers: {
-                    Authorization: `Bearer ${access_token}`,
-                    'Content-Type': 'application/json',
-                },
-            }),
-        3,
-        1000
-    );
 
-    if (response.status === 404) {
-        logger.info(`Zoom - No Zoom user found for student with email ${email}`);
-        return null;
-    } else if (!response.ok) {
-        throw new Error(`Zoom failed to get user: ${response.status} ${await response.text()}`);
+    const zoomFetchUser = (zoomIdOrEmail: string) => {
+        return zoomRetry(
+            () =>
+                fetch(`${zoomUserApiUrl}/${zoomIdOrEmail}`, {
+                    method: 'GET',
+                    headers: {
+                        Authorization: `Bearer ${access_token}`,
+                        'Content-Type': 'application/json',
+                    },
+                }),
+            2,
+            1000
+        );
+    };
+
+    const byEmailResponse = await zoomFetchUser(email);
+    if (byEmailResponse.status === 404) {
+        logger.info(`Zoom - No Zoom user found for Student(${id}) with email ${email}`);
+        if (!zoomUserId) {
+            return null;
+        }
+
+        /** Let's try with the user id because sometimes the zoom API works with one or the other... */
+        const byIdResponse = await zoomFetchUser(zoomUserId);
```

Comment:
Can't we just always fetch by userId?

Reply:
We can, yes. But I'm afraid this will end up in the same issue for other users.
I saw some reports from others on the Zoom support forum that they get 404 for some users when they use the `userId` but not when they use the `email`. So the recommended zoom solution is:
- Can't you find the user by email? Use the id
- Can't you find the user by id? Use the email
🙈

Reply:
Sweet.